src/wovnio/wovnphp/Headers.php

```diff
@@ -252,22 +256,39 @@ private function removeLangFromQuery()
     public function responseOut()
     {
         $urlLanguage = $this->urlLanguage();
+        if (!$urlLanguage || strlen($urlLanguage)==0) {
+            return;
+        }
 
-        if ($urlLanguage && strlen($urlLanguage) > 0) {
-            if (!headers_sent()) {
-                $locationHeaders = array('location', 'Location');
-                $responseHeaders = $this->getResponseHeaders();
+        if (headers_sent()) {
+            return;
+        }
 
-                foreach ($locationHeaders as $locationHeader) {
-                    if (array_key_exists($locationHeader, $responseHeaders)) {
-                        $redirectLocation = $responseHeaders[$locationHeader];
-                        $newLocation = Url::addLangCode($redirectLocation, $this->store, $urlLanguage, $this);
+        $locationHeaders = array('location', 'Location');
+        $responseHeaders = $this->getResponseHeaders();
+        $redirectLocation = null;
+        $newLocation = null;
 
-                        header($locationHeader . ': ' . $newLocation);
-                    }
-                }
+        foreach ($locationHeaders as $locationHeader) {
+            if (array_key_exists($locationHeader, $responseHeaders)) {
+                $redirectLocation = $responseHeaders[$locationHeader];
             }
         }
+
+        if ($redirectLocation) {
+            $newLocation = Url::addLangCode($redirectLocation, $this->store, $urlLanguage, $this);
+        }
+
+        if ($this->shouldRedirect()) {
```

Comment:
i'm confused why this code is here when I just saw in `wovn_interceptor` the same code checking this and sending the redirect?